pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go

```diff
@@ -498,6 +498,21 @@ func (c *Controller) getSelectedWorkloadEntries(ns string, selector map[string]s
 			workloadEntries = append(workloadEntries, wl)
 		}
 	}
+
+	// Include workload entries inlined in service entries (endpoints)
+	allServiceEntries := c.configController.List(gvk.ServiceEntry, ns)
+	for _, se := range allServiceEntries {
+		if labels.Instance(selector).SubsetOf(se.Labels) {
```

Comment:
should we be checking the workload entry endpoints labels instead? (per `spec.labels` https://istio.io/latest/docs/reference/config/networking/workload-entry/#WorkloadEntry)

if empty then perhaps it defaults to SE labels itself? not sure based on docs, should test with sidecar / read the code to make sure we use the right selector here.

Reply:
@kdorosh I'm not sure SE can have labels.
SE can either have a selector of labels or endpoints with labels. SE itself doesn't have labels. Am I wrong?

Reply:
WorkloadEntry authz takes place in VM setups where the sidecar/ztunnel running on the VM enforces the policy.
I tested a VM setup with sidecar, ServiceEntry with endpoints and authz policies that selects labels. The authz is not enforced in such scenarios today and the only way to have policy on suce SE is with global / ns policies (no selector).

Therefore, this scenario is in uncharted water and any behavior will be improvement over sidecar's.
I changed the code to first check the endpoints labels and if not exist try to check the SE's labels. I think it makes sense as a user.

Reply:
Thinking about this more, I don't think we want to inherit SE labels if WE labels are empty. If there's no precedent for that (sidecar), I'd prefer clear semantics rather than potentially confusing ones.

Would love to hear other opinions as well on this